### PR TITLE
Improve error reporting in the "try_run" function and correctly include original command output in the error message

### PR DESCRIPTION
**Pull request by Kami**
This pull request fixes `check_output` function to correctly include output of the command which ran in the `CalledProcessError` exception which is thrown.

In addition to that, it also includes return (exit) code in the printed error message.

Before this change it was very hard / impossible to debug various codecove command failures because the original command output was not included in the error which is printed to the console.

Before (as you can see, output is always `None` which makes troubleshooting practically impossible):

```bash
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.15
==> Detecting CI provider
    Travis Detected
==> Preparing upload
==> Processing gcov (disable by -X gcov)
    Executing gcov (find /home/travis/build/StackStorm/st2 -not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**' -type f -name '*.gcno'  -exec gcov -pb  {} +)
==> Collecting reports
    Mergeing coverage reports
    Error running `coverage combine -a`: None
Error: No coverage report found
Tip: See an example python repo: https://github.com/codecov/example-python
Support channels:
  Email:   hello@codecov.io
  IRC:     #codecov
  Gitter:  https://gitter.im/codecov/support
  Twitter: @codecov
```

After (command output is correctly included which makes troubleshooting possible / easier):

```bash
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.15
==> Detecting CI provider
    Travis Detected
    Fixing merge commit SHA
==> Preparing upload
==> Processing gcov (disable by -X gcov)
    Executing gcov (find /home/travis/build/StackStorm/st2 -not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**' -type f -name '*.gcno'  -exec gcov -pb  {} +)
==> Collecting reports
    Mergeing coverage reports
    Error running `coverage combine -a`: output=Couldn't trace with concurrency=eventlet, the module isn't installed.
, returncode=1
Error: No coverage report found
Tip: See an example python repo: https://github.com/codecov/example-python
Support channels:
  Email:   hello@codecov.io
  IRC:     #codecov
  Gitter:  https://gitter.im/codecov/support
  Twitter: @codecov
```

### CODE REVIEW

File: codecov/__init__.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def check_output(cmd, **popen_args):
     process = Popen(cmd, stdout=PIPE, **popen_args)
     output, _ = process.communicate()
     if process.returncode:
-        raise CalledProcessError(process.returncode, cmd)
+        raise CalledProcessError(process.returncode, cmd, output)
```
Comment (Kami) [inline codecov/__init__.py]: Constructor signature - https://github.com/enthought/Python-2.7.3/blob/master/Lib/subprocess.py#L408

**Comment by Kami** [comment]
Any progress on this?

We still need to use our fork and not upstream version because this change hasn't been accepted / merged upstream yet.

Thanks.

**Comment by hootener** [comment]
I'll bring in @thomasrockhu as point of contact for getting this merged. However it seems like appveyor is failing for every build?

**Comment by thomasrockhu** [comment]
@Kami we are working on getting the `appveyor` tests passing.  We will get this merged ASAP

File: codecov/__init__.py [hunk 2]
```diff
@@ -213,17 +213,18 @@ def check_output(cmd, **popen_args):
...
     else:
         assert process.returncode == 0
         return output.decode("utf-8")
 
 
 def try_to_run(cmd, shell=False, cwd=None):
     try:
-        return check_output(cmd, shell=shell, cwd=cwd)
+        return check_output(cmd, shell=True, cwd=cwd)
```
Comment (thomasrockhu) [inline codecov/__init__.py:224]:
```suggestion
        return check_output(cmd, shell=shell, cwd=cwd)
```

**Review by thomasrockhu — APPROVED** [review]

**Comment by thomasrockhu** [comment]
Thank you for this @Kami!

**Pull request merged**